rasa/utils/tensorflow/models.py

```diff
@@ -574,118 +301,95 @@ def batch_to_model_data_format(
+            # static value
+            if number_of_dimensions > 1 and (
+                batch[idx].shape is None or batch[idx].shape[-1] is None
+            ):
+                shape = [None] * (number_of_dimensions - 1) + [feature_dimension]
+                batch[idx].set_shape(shape)
 
-    def predict_on_batch(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+            return batch[idx], idx + 1
 
-    def fit_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
+        # convert to Tensor
+        return (
+            tf.constant(batch[idx], dtype=tf.float32, shape=batch[idx].shape),
+            idx + 1,
         )
 
-    def evaluate_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+    @staticmethod
+    def _convert_sparse_features(
+        batch: Union[Tuple[tf.Tensor], Tuple[np.ndarray]],
+        feature_dimension: int,
+        idx: int,
+        number_of_dimensions: int,
+    ):
+        # explicitly substitute last dimension in shape with known
+        # static value
+        shape = [batch[idx + 2][i] for i in range(number_of_dimensions - 1)] + [
+            feature_dimension
+        ]
+        return tf.SparseTensor(batch[idx], batch[idx + 1], shape), idx + 3
 
-    def predict_generator(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+    def call(
+        self,
+        inputs: Union[tf.Tensor, List[tf.Tensor]],
+        training: Optional[tf.Tensor] = None,
+        mask: Optional[tf.Tensor] = None,
+    ) -> Union[tf.Tensor, List[tf.Tensor]]:
+        """Calls the model on new inputs.
 
-    def call(self, *args, **kwargs) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+        Arguments:
+            inputs: A tensor or list of tensors.
+            training: Boolean or boolean scalar tensor, indicating whether to run
+              the `Network` in training mode or inference mode.
+            mask: A mask or list of masks. A mask can be
+                either a tensor or None (no mask).
 
-    def get_config(self) -> None:
-        raise Exception(
-            "This method should neither be called nor implemented in our code."
-        )
+        Returns:
+            A tensor if there is a single output, or
+            a list of tensors if there are more than one outputs.
+        """
+        # This method needs to be implemented, otherwise the super class is raising a
+        # NotImplementedError('When subclassing the `Model` class, you should
+        #   implement a `call` method.')
+        pass
```

Comment:
What do you think about explicitly raising `NotImplementedError` instead of this comment? (basically the same way as we do it for other methods in `Model` class and its subclasses)

Reply:
why would we raise NotImplementedError? it is specifically overwritten not to raise this error

Reply:
We cannot raise a `NotImplementedError` as training would fail in that case. We explicitly want to do nothing here. That is the reason we overwrite this method.

Reply:
Oh okay, that makes sense then 👍